lib/services/test-execution-service.ts

```diff
@@ -122,8 +145,36 @@ class TestExecutionService implements ITestExecutionService {
 			if (this.$options.debugBrk) {
 				karmaConfig.browserNoActivityTimeout = 1000000000;
 			}
+
+			karmaConfig.projectDir = this.$projectData.projectDir;
 			this.$logger.debug(JSON.stringify(karmaConfig, null, 4));
-			new KarmaServer(karmaConfig).start();
+
+			let karmaRunner = require("child_process").fork(path.join(__dirname, "karma-execution.js"));
+			karmaRunner.send({karmaConfig: karmaConfig});
+			karmaRunner.on("message", (karmaData: any) => {
+				fiberBootstrap.run(() => {
+					this.$logger.trace("## Unit-testing: Parent process received message", karmaData);
+					let port: string;
+					if(karmaData.url) {
+						port = karmaData.url.port;
+						let socketIoJsUrl = `http://localhost:${port}/socket.io/socket.io.js`;
+						let socketIoJs = this.$httpClient.httpRequest(socketIoJsUrl).wait().body;
+						this.$fs.writeFile(path.join(projectDir, TestExecutionService.SOCKETIO_JS_FILE_NAME), socketIoJs).wait();
+					}
+
+					if(karmaData.launcherConfig) {
+						let configOptions: IKarmaConfigOptions = JSON.parse(karmaData.launcherConfig);
+						let configJs = this.generateConfig(port, configOptions);
+						this.$fs.writeFile(path.join(projectDir, TestExecutionService.CONFIG_FILE_NAME), configJs).wait();
+					}
+
+					if(this.$options.debugBrk) {
+						this.getDebugService(platform).debug().wait();
```

Comment:
I'm not sure I've understood your point here, but I've moved the generation of LiveSync data and calling LiveSync service in a separate method that is called only in the `else` statement.